src/callback_tracking.jl

```diff
@@ -214,6 +215,12 @@ function _setup_reverse_callbacks(cb::Union{ContinuousCallback,DiscreteCallback,
         end
 
         update_p = copy_to_integrator!(cb,y,integrator.p,integrator.t,indx,pos_neg)
+        # reshape u and du (y and dy) to match forward pass (e.g., for matrices as initial conditions). Only needed for BacksolveAdjoint
+        if sensealg isa BacksolveAdjoint
+          _size = pos_neg ? size(cb.affect!.uleft[indx]) : size(cb.affect_neg!.uleft[indx])
```

Comment:
@ChrisRackauckas, no it's not superseded by #580. This addition reshaping the state for matrix-valued initial conditions (#555) is independent.